src/main/java/it/gov/pagopa/ranking/service/RankingRequestsApiServiceImpl.java

```diff
@@ -53,4 +68,40 @@ public List<RankingRequestsApiDTO> findByInitiativeId(String organizationId, Str
             return out;
         }
     }
+
+    @Override
+    public RankingPageDTO findByInitiativeIdPaged(String organizationId, String initiativeId, int page, int size, String beneficiaryRankingStatus) {
+
+        InitiativeConfig initiative = rankingContextHolderService.getInitiativeConfig(initiativeId, organizationId);
+        if (initiative == null) {
+            return null;
+        } else {
+            List<RankingRequestsApiDTO> rankingDtoList = new ArrayList<>();
```

Comment:
instead of initialize the out as a ArrayList, let's not initialize it.
Inside the if let's fill it using the stream api (requests.stream().map(....)
in the else, let's return the Collections.emptyList

This way, we could have a performance boost in the population of the lists in case of large size requested